src/main.cc

```diff
@@ -178,6 +178,39 @@ signal_cb(ev_loop *loop, struct ev_signal *w, int revents)
 	tarantool_exit(0);
 }
 
+static void
+signal_sigint_cb(ev_loop *loop, struct ev_signal *w, int revents)
+{
+	(void) loop;
+	(void) w;
+	(void) revents;
+
+	/**
+	 * If running in daemon mode, tarantool exits on SIGINT.
+	 * Complain about possibly sudden and unexpected death.
+	 */
+	if (pid_file) {
+		say_crit("got signal %d - %s", w->signum, strsignal(w->signum));
+		tarantool_exit(0);
+		return;
+	}
+
+	/**
+	 * Setting prompt explicitly every time, cause of
+	 * need to return from search mode.
+	 */
+	rl_set_prompt("tarantool> ");
+
+	const char *line_end = "^C\n";
+	ssize_t rc = write(STDOUT_FILENO, line_end, strlen(line_end));
+	(void) rc;
+
+	RL_UNSETSTATE(RL_STATE_ISEARCH | RL_STATE_NSEARCH | RL_STATE_SEARCH);
+	rl_on_new_line();
+	rl_replace_line("", 0);
+	rl_redisplay();
```

Comment:
The feature doesn't work with a remote console:

```
tarantool> require('console').connect('localhost:3302')
connected to localhost:3302
---
- true
...

localhost:3302> ^C
tarantool>
```

Reply:
Fixed.